## Первое задание

In [1]:
import pandas as pd
import numpy as np

In [3]:
df = pd.read_csv('tips.csv')
df.head(5)

total_bill   tip     sex smoker  day    time  size  price_per_person  \
0       16.99  1.01  Female     No  Sun  Dinner     2              8.49   
1       10.34  1.66    Male     No  Sun  Dinner     3              3.45   
2       21.01  3.50    Male     No  Sun  Dinner     3              7.00   
3       23.68  3.31    Male     No  Sun  Dinner     2             11.84   
4       24.59  3.61  Female     No  Sun  Dinner     4              6.15   

           Payer Name         CC Number Payment ID  
0  Christy Cunningham  3560325168603410    Sun2959  
1      Douglas Tucker  4478071379779230    Sun4608  
2      Travis Walters  6011812112971322    Sun4458  
3    Nathaniel Harris  4676137647685994    Sun5260  
4        Tonya Carter  4832732618637221    Sun2251

In [6]:
df_num = df[['total_bill', 'tip', 'size', 'price_per_person', 'CC Number']]
df_num

total_bill   tip  size  price_per_person         CC Number
0         16.99  1.01     2              8.49  3560325168603410
1         10.34  1.66     3              3.45  4478071379779230
2         21.01  3.50     3              7.00  6011812112971322
3         23.68  3.31     2             11.84  4676137647685994
4         24.59  3.61     4              6.15  4832732618637221
..          ...   ...   ...               ...               ...
239       29.03  5.92     3              9.68  5296068606052842
240       27.18  2.00     2             13.59  3506806155565404
241       22.67  2.00     2             11.34  6011891618747196
242       17.82  1.75     2              8.91     4375220550950
243       18.78  3.00     2              9.39  3511451626698139

[244 rows x 5 columns]

## Второе задание

In [8]:
df_num.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 5 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   total_bill        244 non-null    float64
 1   tip               244 non-null    float64
 2   size              244 non-null    int64  
 3   price_per_person  244 non-null    float64
 4   CC Number         244 non-null    int64  
dtypes: float64(3), int64(2)
memory usage: 9.7 KB


In [9]:
df_num.describe()

total_bill         tip        size  price_per_person     CC Number
count  244.000000  244.000000  244.000000        244.000000  2.440000e+02
mean    19.785943    2.998279    2.569672          7.888197  2.563496e+15
std      8.902412    1.383638    0.951100          2.914234  2.369340e+15
min      3.070000    1.000000    1.000000          2.880000  6.040679e+10
25%     13.347500    2.000000    2.000000          5.800000  3.040731e+13
50%     17.795000    2.900000    2.000000          7.255000  3.525318e+15
75%     24.127500    3.562500    3.000000          9.390000  4.553675e+15
max     50.810000   10.000000    6.000000         20.270000  6.596454e+15

**total_bill:**
- непустых значений: 244
- тип данных: float64
- Min: 3.070000	
- Max: 50.810000
- Столбец сожержит общий чек на один стол в независимости от того, сколько людей там сидело. 

**tip:**
- Непустые значения: 244
- Тип данных: float64
- Max: 10.000000
- Min: 1.000000
- Столбец содержит сумму чаевых, которую оставил клиент.

**size:**
- Непустые значения: 244
- Тип данных: int64
- Max: 6.000000
- Min: 1.000000
- Столбец содержит количество людей сделавших один заказ.

**price_per_person:**
- Непустые значения: 244
- Тип данных: float64
- Max: 20.270000
- Min: 2.880000
- Столбец содержит информацию о чеке на каждого человека в независимости от того, пришел ли он группой или один. 

**CC Number:**
- Непустые значения: 244
- Тип данных: int64
- Max: 
- Min: 
- Столбец содержит номера карт клиентов, либо номер заказа. Поэтому выводить минимальное и максимальное для данной перменной нет смысла.

## Третье задание 

In [10]:
df_num['after_dot'] = df_num['total_bill'].apply(lambda x: round(x % 1, 2))

C:\Users\azizr\AppData\Local\Temp\ipykernel_13332\780531956.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_num['after_dot'] = df_num['total_bill'].apply(lambda x: round(x % 1, 2))


In [17]:
df_num.head(3)

total_bill  tip  size  price_per_person         CC Number  after_dot  \
0        17.0  1.0     2              8.49  3560325168603410       0.99   
1        10.0  2.0     3              3.45  4478071379779230       0.34   
2        21.0  4.0     3              7.00  6011812112971322       0.01   

   percent  sum_bill_tip  
0     5.88          18.0  
1    20.00          12.0  
2    19.05          25.0

In [13]:
df_num['total_bill'] = df_num['total_bill'].round()
df_num['tip'] = df_num['tip'].round()

C:\Users\azizr\AppData\Local\Temp\ipykernel_13332\27693488.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_num['total_bill'] = df_num['total_bill'].round()
C:\Users\azizr\AppData\Local\Temp\ipykernel_13332\27693488.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_num['tip'] = df_num['tip'].round()


In [15]:
df_num['percent'] = round(df_num['tip'] / df_num['total_bill'] * 100, 2)

C:\Users\azizr\AppData\Local\Temp\ipykernel_13332\4131909993.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_num['percent'] = round(df_num['tip'] / df_num['total_bill'] * 100, 2)


In [16]:
df_num['sum_bill_tip'] = df_num['total_bill'] + df_num['tip']

C:\Users\azizr\AppData\Local\Temp\ipykernel_13332\1968569920.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_num['sum_bill_tip'] = df_num['total_bill'] + df_num['tip']


In [19]:
df_num.head(3)

total_bill  tip  size  price_per_person         CC Number  after_dot  \
0        17.0  1.0     2              8.49  3560325168603410       0.99   
1        10.0  2.0     3              3.45  4478071379779230       0.34   
2        21.0  4.0     3              7.00  6011812112971322       0.01   

   percent  sum_bill_tip  
0     5.88          18.0  
1    20.00          12.0  
2    19.05          25.0

## Четвертое задание

In [22]:
df_str = df[['sex', 'smoker', 'day', 'time', 'Payer Name']]
df_str.head(3)

sex smoker  day    time          Payer Name
0  Female     No  Sun  Dinner  Christy Cunningham
1    Male     No  Sun  Dinner      Douglas Tucker
2    Male     No  Sun  Dinner      Travis Walters

## Пятое задание 

In [27]:
list_names = []
for payer_name in df_str['Payer Name']:
  list_names.append(payer_name.split(' ')[0])

df_str['name'] = list_names


C:\Users\azizr\AppData\Local\Temp\ipykernel_13332\4094280119.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_str['name'] = list_names


In [30]:
df_str.head(2)

sex smoker  day    time          Payer Name     name     surname
0  Female     No  Sun  Dinner  Christy Cunningham  Christy  Cunningham
1    Male     No  Sun  Dinner      Douglas Tucker  Douglas      Tucker

In [29]:
list_surnames = []
for payer_name in df_str['Payer Name']:
  list_surnames.append(payer_name.split(' ')[1])

df_str['surname'] = list_surnames

C:\Users\azizr\AppData\Local\Temp\ipykernel_13332\4043933457.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_str['surname'] = list_surnames
